src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/TelemetryReporterTests.cs

```diff
@@ -20,6 +20,9 @@ private async Task<ITelemetryReporter> CreateReporterAsync()
     {
         var exportProvider = await LanguageServerTestComposition.CreateExportProviderAsync(TestOutputLogger.Factory, includeDevKitComponents: true, out var _);
 
+        // VS Telemetry requires this environment variable to be set.
+        Environment.SetEnvironmentVariable("CommonPropertyBagPath", Path.GetTempFileName());
```

Comment:
>Environment.SetEnvironmentVariable("CommonPropertyBagPath", Path.GetTempFileName()); [](http://example.com/codeflow?start=8&length=84)

We don't have to do this in product code?

I see where the vstelemetryapi code uses this value (unfortunate they don't expose a constant for the name), do we actually use the file anywhere? 

Would it make sense for them not to fail if they can't find the env variable, but to instead just set it to Path.GetTempFileName() instead?

Reply:
> We don't have to do this in product code?

We do do this in product code (it happens here - https://github.com/dotnet/vscode-csharp/blob/dd7c7f297ae76babedc3c036e6face93ed960f54/src/lsptoolshost/roslynLanguageServer.ts#L847)

cc @jonathanjyi

Reply:
Hi @ToddGrun, that file is used to pass specific values from VS Code (Extension version, telemetry level, etc) over to the TelemetryAPI that we don't want to pass on the command line (because we're trying to limit doing that). If that file is not found, then telemetry does error out on the managed side. There's not really anything in that file that other teams would need that you can't get out of TelemetrySession, but if you want to do something with it send me a ping.

Reply:
OK, makes sense. Thanks!